enigma-principal/enclave/src/epoch_keeper_t/epoch_t.rs

```diff
@@ -25,32 +27,25 @@ impl Epoch {
             .get_selected_worker(sc_addr, self.seed)
             .ok_or_else(|| SystemError(EnclaveSystemError::WorkerAuthError { err: "Worker selection returns nothing.".to_string() }))
     }
-}
 
-impl RawEncodable for Epoch {
-    /// Encode the Epoch as Ethereum ABI parameters
-    fn raw_encode(&self) -> Bytes {
-        let raw_seed = H256::from(self.seed).0.to_vec();
-        let mut image = raw_seed.len().to_be_bytes().to_vec();
-        image.extend(raw_seed);
-
-        let raw_nonce = H256::from(self.nonce).0.to_vec();
-        image.extend(raw_nonce.len().to_be_bytes().to_vec());
-        image.extend(raw_nonce);
-
-        image.extend(self.worker_params.workers.len().to_be_bytes().to_vec());
-        for addr in self.worker_params.workers.clone() {
-            let raw_addr = addr.0.to_vec();
-            image.extend(raw_addr.len().to_be_bytes().to_vec());
-            image.extend(raw_addr);
-        }
-
-        image.extend(self.worker_params.stakes.len().to_be_bytes().to_vec());
-        for amount in self.worker_params.stakes.clone() {
-            let raw_amount = H256::from(amount).0.to_vec();
-            image.extend(raw_amount.len().to_be_bytes().to_vec());
-            image.extend(raw_amount);
-        }
-        image
+    pub fn encode_epoch(&self) -> Bytes {
+        let mut encoding_len: usize = 0;
+        let mut encoding: Vec<u8> = Vec::new();
+
+        let (len_seed, seed_encoding) = self.seed.hash_encode();
+        let (len_nonce, nonce_encoding) = self.nonce.hash_encode();
+        let (len_workers, workers_encoding) = self.worker_params.workers.hash_encode();
+        let (len_stakes, stakes_encoding) = self.worker_params.stakes.hash_encode();
```

Comment:
`self.worker_params.block_number` is not encoded.

Reply:
it wasn't added in the first place, so I havent added it here as well. @fredfortier did you know why the block_number is not included in the encoding?